client/signup/steps/website-content/index.tsx

```diff
@@ -84,19 +84,17 @@ function WebsiteContentStep( {
 	const translatedPageTitles = useTranslatedPageTitles();
 
 	useEffect( () => {
-		if ( websiteContent.pages.length > 0 ) {
-			// Already initialized.
-			return;
-		}
-
```

Comment:
I didn't test this but, Doesn't this clear the saved content every-time we land on this page? The "Saving" concept is completely removed here right?

I think a better quick fix is to save the current active siteId and clear the pages whenever there is a different site.
So we cache page details only one site per person (For now).

Reply:
CC @aneeshd16

Reply:
Yes, good catch! An alternative is to not clear the text content in any case. I cannot think of a real-world use case where we would want to reset the text input. This is the change I am suggesting:
<img width="1133" alt="image" src="https://user-images.githubusercontent.com/5436027/196634339-d42dd1c0-8061-4180-9665-f2f10b0fc9a3.png">

WDYT?

Reply:
Looking at the reducer this is not how it works (Sorry I just noticed it). Persisted data overrides the initialisation data. The page state does not get cleared by this call if saved data exists. 

https://github.com/Automattic/wp-calypso/blob/0f42dd2f2b4cd501fb1707af2af49a554aaca12c/client/state/signup/steps/website-content/reducer.ts#L32-L45

I know we are trying to get a quick fix in here but trying to evade this logic without clearly tackling it might lead to even more bugs.

Simply we need to build this fix on top of the logic where **state is cleared IFF the site id for this page changes**. This can easily be done by maintaining the current siteId context as a state variable.

If you want I can have a crack at it because I am pretty familiar with the area and can get something out quickly.

The actual long term fix would of course be a keyed state where multiple site wise, website content contexts can be maintained.